yt_dlp/extractor/vimeo.py

```diff
@@ -612,6 +651,19 @@ def _extract_url(url, webpage):
         urls = VimeoIE._extract_urls(url, webpage)
         return urls[0] if urls else None
 
+    @classmethod
+    def suitable(cls, url):
+        """
+        as vimeo URLs are bit tricky we make sure, that the
+        _VALID_URL of the classes don't overlap
+        this makes regex definitions much simpler
+        """
+        exclude_regex = []
+        for child in VimeoBaseInfoExtractor.__subclasses__():
+            if cls != child and hasattr(child, '_VALID_URL'):
+                exclude_regex.append(child._VALID_URL)
+        return super().suitable(url) and not any(re.match(regex, url) for regex in exclude_regex)
```

Comment:
I get your idea, but isn't it easier with just a complex regex?

Reply:
You know what they say about regular expressions? :-)
I couldn't come up with a pattern. And maintenance can be painful. Here are just some examples:
`https://vimeo.com/album/3253534 [vimeo:album]`
`https://vimeo.com/channels/tributes [vimeo:channel]` 
`https://vimeo.com/nkistudio/videos [vimeo:user]`
`https://vimeo.com/groups/meetup [vimeo:group]`
`https://vimeo.com/user21297594/review/75524534/3c257a1b5d [vimeo:review]`
`https://vimeo.com/channels/tributes/6213729 [vimeo]`

I prefer only one class per platform that takes all urls and then does a dispatch based on the pattern deciding if it is a link to a single video or a playlist...